### PR TITLE
[CP Auth 1/5]: Add environment configuration

### PR DESCRIPTION
**Pull request by axbarsan**
Part of the larger https://github.com/giantswarm/happa/pull/1547

Includes environment configuration, such as env variables, deployment configuration, local proxy settings etc.

### CODE REVIEW

File: scripts/webpack/proxyPlugin.js [hunk 1]
```diff
@@ -117,7 +117,13 @@ class ProxyPlugin {
 
       req
         .pipe(
-          request(cleanProxyUrl + req.url).on('response', (response) => {
+          request({
+            url: cleanProxyUrl + req.url,
+            rejectUnauthorized: false,
+            followAllRedirects: false,
+            followRedirect: false,
+            followOriginalHttpMethod: true,
```
Comment (axbarsan) [inline scripts/webpack/proxyPlugin.js:125]: Necessary to support login redirects after OIDC login. Doesn't break anything else (so far)

**Review by ubergesundheit — APPROVED** [review]
Excellent

File: webpack.common.js [hunk 2]
```diff
@@ -48,30 +48,57 @@ const determineAudienceURL = () => {
   return apiAudienceUrl;
 };
 
+function makeCPApiEndpointURL(currentValue, apiEndpoint) {
```
Comment (oponder) [inline webpack.common.js:51]: What are these functions for? I don't really follow what they're doing.

Like why do we need to have this logic?

Comment (axbarsan) [reply webpack.common.js:51]: This helps derive the CP API URL from the regular API endpoint URL, if the API endpoint set is pointing to an installation, so you wouldn't have to set the CP API URL manually

**Pull request merged**